File: android/app/src/main/java/com/buttonlog/app/data/model/Team.kt
package com.buttonlog.app.data.model

import com.google.gson.annotations.SerializedName

data class Team(
    val id: String,
    val name: String,
    val description: String? = null,
    val icon: String = "people",
    val color: String = "#3B82F6",
    @SerializedName("owner_id")
    val ownerId: String,
    @SerializedName("organization_id")
    val organizationId: String? = null,
    @SerializedName("member_count")
    val memberCount: Int? = null,
    @SerializedName("button_count")
    val buttonCount: Int? = null,
    @SerializedName("created_at")
    val createdAt: String,
    @SerializedName("updated_at")
    val updatedAt: String,
    val owner: PublicUser? = null,
    @SerializedName("my_role")
    val myRole: String? = null,
    @SerializedName("can_manage")
    val canManage: Boolean? = null,
    val members: List<TeamMember>? = null,
    val buttons: List<TeamButton>? = null
)

data class TeamMember(
    val id: String,
    @SerializedName("user_id")
    val userId: String,
    @SerializedName("team_id")
    val teamId: String,
    val role: String,
    @SerializedName("joined_at")
    val joinedAt: String,
    @SerializedName("invited_by_id")
    val invitedById: String? = null,
    val user: PublicUser? = null,
    @SerializedName("invited_by")
    val invitedBy: PublicUser? = null
)

data class TeamButton(
    val id: String,
    @SerializedName("team_id")
    val teamId: String,
    @SerializedName("button_id")
    val buttonId: String,
    val permission: String,
    @SerializedName("added_by_id")
    val addedById: String? = null,
    @SerializedName("added_at")
    val addedAt: String,
    val button: TeamButtonInfo? = null
)

data class TeamButtonInfo(
    val id: String,
    val name: String,
    val type: String,
    val icon: String,
    val color: String
)

data class TeamInvitation(
    val id: String,
    @SerializedName("team_id")
    val teamId: String,
    @SerializedName("inviter_id")
    val inviterId: String,
    @SerializedName("invitee_id")
    val inviteeId: String,
    val role: String,
    val status: String,
    @SerializedName("created_at")
    val createdAt: String,
    @SerializedName("expires_at")
    val expiresAt: String? = null,
    val team: TeamSummary? = null,
    val inviter: PublicUser? = null,
    val invitee: PublicUser? = null
)

data class TeamSummary(
    val id: String,
    val name: String,
    val description: String? = null,
    val icon: String,
    val color: String
)

data class TeamsResponse(
    val teams: List<Team>,
    @SerializedName("pending_invitations")
    val pendingInvitations: List<TeamInvitation>
)

data class CreateTeamRequest(
    val name: String,
    val description: String? = null,
    val icon: String = "people",
    val color: String = "#3B82F6"
)

File: iphone/ButtonLog/Models/Team.swift
import Foundation

// MARK: - Team Model

struct Team: Identifiable, Codable {
    let id: String
    let name: String
    let description: String?
    let icon: String
    let color: String
    let ownerId: String
    let organizationId: String?
    let memberCount: Int?
    let buttonCount: Int?
    let createdAt: Date
    let updatedAt: Date

    // Optional associations
    var owner: PublicUser?
    var myRole: TeamRole?
    var canManage: Bool?
    var members: [TeamMember]?
    var buttons: [TeamButton]?

    enum CodingKeys: String, CodingKey {
        case id, name, description, icon, color
        case ownerId = "owner_id"
        case organizationId = "organization_id"
        case memberCount = "member_count"
        case buttonCount = "button_count"
        case createdAt = "created_at"
        case updatedAt = "updated_at"
        case owner
        case myRole = "my_role"
        case canManage = "can_manage"
        case members, buttons
    }
}

// MARK: - Team Role

enum TeamRole: String, Codable, CaseIterable {
    case owner = "owner"
    case admin = "admin"
    case member = "member"

    var displayName: String {
        switch self {
        case .owner: return "Owner"
        case .admin: return "Admin"
        case .member: return "Member"
        }
    }

    var isAdmin: Bool {
        self == .owner || self == .admin
    }
}

// MARK: - Team Member

struct TeamMember: Identifiable, Codable {
    let id: String
    let userId: String
    let teamId: String
    let role: TeamRole
    let joinedAt: Date
    let invitedById: String?
    var user: PublicUser?
    var invitedBy: PublicUser?

    enum CodingKeys: String, CodingKey {
        case id
        case userId = "user_id"
        case teamId = "team_id"
        case role
        case joinedAt = "joined_at"
        case invitedById = "invited_by_id"
        case user
        case invitedBy = "invited_by"
    }
}

// MARK: - Team Button

struct TeamButton: Identifiable, Codable {
    let id: String
    let teamId: String
    let buttonId: String
    let permission: TeamButtonPermission
    let addedById: String?
    let addedAt: Date
    var button: TeamButtonInfo?

    enum CodingKeys: String, CodingKey {
        case id
        case teamId = "team_id"
        case buttonId = "button_id"
        case permission
        case addedById = "added_by_id"
        case addedAt = "added_at"
        case button
    }
}

struct TeamButtonInfo: Identifiable, Codable {
    let id: String
    let name: String
    let type: String
    let icon: String
    let color: String
}

enum TeamButtonPermission: String, Codable, CaseIterable {
    case view = "view"
    case click = "click"
    case admin = "admin"

    var displayName: String {
        switch self {
        case .view: return "View Only"
        case .click: return "Can Click"
        case .admin: return "Full Access"
        }
    }
}

// MARK: - Team Invitation

struct TeamInvitation: Identifiable, Codable {
    let id: String
    let teamId: String
    let inviterId: String
    let inviteeId: String
    let role: TeamRole
    let status: InvitationStatus
    let createdAt: Date
    let expiresAt: Date?
    var team: TeamSummary?
    var inviter: PublicUser?
    var invitee: PublicUser?

    enum CodingKeys: String, CodingKey {
        case id
        case teamId = "team_id"
        case inviterId = "inviter_id"
        case inviteeId = "invitee_id"
        case role, status
        case createdAt = "created_at"
        case expiresAt = "expires_at"
        case team, inviter, invitee
    }

    var isExpired: Bool {
        if let expiresAt = expiresAt {
            return expiresAt < Date()
        }
        return false
    }
}

struct TeamSummary: Identifiable, Codable {
    let id: String
    let name: String
    let description: String?
    let icon: String
    let color: String
}

enum InvitationStatus: String, Codable {
    case pending = "pending"
    case accepted = "accepted"
    case declined = "declined"
    case cancelled = "cancelled"
    case expired = "expired"

    var displayName: String {
        switch self {
        case .pending: return "Pending"
        case .accepted: return "Accepted"
        case .declined: return "Declined"
        case .cancelled: return "Cancelled"
        case .expired: return "Expired"
        }
    }
}

// MARK: - Create Team Request

struct CreateTeamRequest {
    var name: String = ""
    var description: String = ""
    var icon: String = "people"
    var color: String = "#3B82F6"

    var isValid: Bool {
        !name.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty
    }

    func toRequestBody() -> [String: Any] {
        var body: [String: Any] = [
            "name": name.trimmingCharacters(in: .whitespacesAndNewlines),
            "icon": icon,
            "color": color
        ]

        let trimmedDescription = description.trimmingCharacters(in: .whitespacesAndNewlines)
        if !trimmedDescription.isEmpty {
            body["description"] = trimmedDescription
        }

        return body
    }
}
